### PR TITLE
PEP8ify some variable names.

### PR DESCRIPTION
**Pull request by anntzer**
nrows/ncols is already used as parameter names in subplots() so
standardizing on them seems good enough.

Note that the docstring of SubplotBase was previously incorrect:
one didn't need `numRows <= numCols <= plotNum < 10`, just that all
three be single-digit.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axes/_subplots.py [hunk 1]
```diff
@@ -17,16 +17,19 @@ class SubplotBase:
...
-        being created.  *plotNum* starts at 1 in the upper left
-        corner and increases to the right.
-
-        If *numRows* <= *numCols* <= *plotNum* < 10, *args* can be the
-        decimal integer *numRows* * 100 + *numCols* * 10 + *plotNum*.
+        Parameters
+        ----------
+        fig : `matplotlib.figure.Figure`
+
+        *args : tuple (*nrows*, *ncols*, *index*)
```
Comment (timhoffm) [inline lib/matplotlib/axes/_subplots.py]:
```suggestion
        *args : tuple (*nrows*, *ncols*, *index*) or int
```
I think that should be mentioned, even though the int has very special semantics.

File: lib/mpl_toolkits/axes_grid1/axes_divider.py [hunk 2]
```diff
@@ -350,15 +350,17 @@ def __init__(self, fig, *args, horizontal=None, vertical=None,
...
-            The array of subplots in the figure has dimensions *numRows*,
-            *numCols*, and *plotNum* is the number of the subplot
-            being created.  *plotNum* starts at 1 in the upper left
-            corner and increases to the right.
-
-            If *numRows* <= *numCols* <= *plotNum* < 10, *args* can be the
-            decimal integer *numRows* * 100 + *numCols* * 10 + *plotNum*.
+        fig : `matplotlib.figure.Figure`
+
+        *args : tuple (*nrows*, *ncols*, *index*)
```
Comment (timhoffm) [inline lib/mpl_toolkits/axes_grid1/axes_divider.py]:
```suggestion
        *args : tuple (*nrows*, *ncols*, *index*) or int
```

File: lib/matplotlib/axes/_subplots.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/axes/_subplots.py]: done

File: lib/mpl_toolkits/axes_grid1/axes_divider.py (hunk 2 above)
Comment (anntzer) [reply lib/mpl_toolkits/axes_grid1/axes_divider.py]: done

**Pull request merged**